src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateEntry.cs

```diff
@@ -23,7 +24,17 @@ public abstract class ModelStateEntry
         /// <summary>
         /// Gets the <see cref="ModelErrorCollection"/> for this entry.
         /// </summary>
-        public ModelErrorCollection Errors { get; } = new ModelErrorCollection();
+        public ModelErrorCollection Errors
+        {
+            get
+            {
+                if(_errors == null)
+                {
+                    _errors = new ModelErrorCollection();
+                }
+                return _errors;
+            }
+        }
```

Comment:
@dougbu @rynowak I ran a test of 1500 POST calls with JSON payloads ranging between 14 KB and 500 KB. Here's the instance count of types with the text "Error" in it, before the change:

![image](https://cloud.githubusercontent.com/assets/950290/14436404/60174350-ffd1-11e5-9170-87b82407d1d7.png)

This is what I see after the change:

![image](https://cloud.githubusercontent.com/assets/950290/14436412/69c537ae-ffd1-11e5-8fff-e9de952c05ea.png)

I'm currently working to get this test into the Performance repository.